src/edu/cmu/cs/sasylf/Proof.java

```diff
@@ -157,69 +209,526 @@ public int getWarningCount() {
+	public static String getConclusions() { return conclusions.toString(); }
+
 	private void doParseAndCheck(ModuleFinder mf, Reader r) {
 		FreeVar.reinit();
 		try {
-			syntaxTree = DSLToolkitParser.read(filename,r);
+			syntaxTree = DSLToolkitParser.read(filename, r);
 		} catch (ParseException e) {
 			final TokenSpan errorSpan = new TokenSpan(e.currentToken.next);
-			if (e.expectedTokenSequences != null && e.expectedTokenSequences.length == 1) {
+			if (e.expectedTokenSequences != null &&
+					e.expectedTokenSequences.length == 1) {
 				String expected = e.tokenImage[e.expectedTokenSequences[0][0]];
-				ErrorHandler.recoverableError(Errors.PARSE_EXPECTED, expected, errorSpan);
+				ErrorHandler.recoverableError(Errors.PARSE_EXPECTED, expected,
+																			errorSpan);
 			} else {
 				// do not use "e.getMessage()": not localized
 				ErrorHandler.recoverableError(Errors.PARSE_ERROR, errorSpan);
 			}
 		} catch (TokenMgrError e) {
-			ErrorHandler.recoverableError(Errors.LEXICAL_ERROR, ErrorHandler.lexicalErrorAsLocation(filename, e.getMessage()));
+			ErrorHandler.recoverableError(
+					Errors.LEXICAL_ERROR,
+					ErrorHandler.lexicalErrorAsLocation(filename, e.getMessage()));
 		} catch (RuntimeException e) {
 			e.printStackTrace();
-			final Span errorSpan = new Location(filename,0,0);
-			ErrorHandler.recoverableError(Errors.INTERNAL_ERROR, "Internal error during parsing: " + e, errorSpan);
+			final Span errorSpan = new Location(filename, 0, 0);
+			ErrorHandler.recoverableError(Errors.INTERNAL_ERROR,
+																		"Internal error during parsing: " + e,
+																		errorSpan);
+		}
+		Context ctx = null;
+		try {
+			if (mf != null) ctx = new Context(new ResourceModuleFinder(), syntaxTree);
+			else {
+				mf.setCurrentPackage(id == null ? ModuleFinder.EMPTY_PACKAGE
+																				: id.packageName);
+				ctx = new Context(mf, syntaxTree);
+			}
+		} catch (SASyLFError ex) {
+			// muffle: handled already
+		} catch (RuntimeException ex) {
+			ex.printStackTrace();
+			ErrorHandler.recoverableError(Errors.INTERNAL_ERROR,
+																		ex.getLocalizedMessage(), null);
 		}
-		duringParse = ErrorHandler.getReports().size();
 		if (syntaxTree != null) {
 			try {
 				if (mf == null) syntaxTree.typecheck(new ResourceModuleFinder(), null);
 				else {
-					mf.setCurrentPackage(id == null ? ModuleFinder.EMPTY_PACKAGE : id.packageName);
-					syntaxTree.typecheck(mf,id);
+					mf.setCurrentPackage(id == null ? ModuleFinder.EMPTY_PACKAGE
+																					: id.packageName);
+					syntaxTree.typecheck(mf, id);
 				}
 			} catch (SASyLFError ex) {
 				// muffle: handled already
 			} catch (RuntimeException ex) {
 				ex.printStackTrace();
-				ErrorHandler.recoverableError(Errors.INTERNAL_ERROR, ex.getLocalizedMessage(), null);
+				ErrorHandler.recoverableError(Errors.INTERNAL_ERROR,
+																			ex.getLocalizedMessage(), null);
+			}
+		}
+
```

Comment:
I don't understand this.  It seems to do a bunch of low-level (AST level) SASyLF stuff.  Is this not done somewhere else?  WHat is its purpose?